crunchy-cli-core/src/cli/archive.rs

```diff
@@ -414,7 +414,7 @@ async fn formats_from_series(
             let (ref mut formats, subtitles) = result
                 .entry(season.metadata.season_number)
                 .or_insert_with(BTreeMap::new)
-                .entry(episode.metadata.episode_number)
+                .entry((episode.metadata.sequence_number * 100.0) as u32)
```

Comment:
Instead of multiplying and converting to a u32, changing the type of the BTreeMap from `BTreeMap<u32, BTreeMap<u32, (Vec<Format>, Vec<Subtitle>)>>` to `BTreeMap<u32, BTreeMap<f32, (Vec<Format>, Vec<Subtitle>)>>` and using `episode.metadata.sequence_number` should be fine too(?). Might be better as the multiplying part is then not needed anymore and can't confuse people why it is there in the first place.

Reply:
yeah I wish it was straightforward, but BTreeMap can't have f32 keys without adding something like [OrderedFloat](https://docs.rs/ordered-float/latest/ordered_float/struct.OrderedFloat.html) (presumably due to floating point precision problems)

```
note: the following trait bounds were not satisfied: `f32: Ord`
```